src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -42,6 +42,16 @@ public function __construct(ContainerInterface $container, $resource, array $opt
 
         $this->resource = $resource;
         $this->context = $context ?: new RequestContext();
+
+        if (array_key_exists('charset', $options)) {
+            try {
+                $this->setOption('charset', $options['charset']);
+            } catch (\InvalidArgumentException $e) {
+                // symfony/routing version 3.x does not support the charset option.
```

Comment:
Yes. Because the regexp generated by RouteCompiler now has the `u` flag, this is a BC-breaking change. See my question regarding that issue [here](https://github.com/symfony/symfony/pull/19562#issuecomment-238096051).